providers/google/src/airflow/providers/google/cloud/hooks/gen_ai.py

```diff
@@ -194,3 +197,182 @@ def create_cached_content(
+        list_batch_jobs_config: ListBatchJobsConfig | dict | None = None,
+    ):
+        """
+        Get list of batch jobs using Gemini Batch API.
+
+        :param list_batch_jobs_config: Optional. Configuration of returned iterator.
+        """
+        client = self.get_genai_client()
+        resp = client.batches.list(
+            config=list_batch_jobs_config,
+        )
+        return resp
+
+    def create_batch_job(
+        self,
+        model: str,
+        source: list | str,
+        create_batch_job_config: CreateBatchJobConfig | dict | None = None,
+    ) -> BatchJob:
+        """
+        Create batch job using Gemini Batch API to process large-scale, non-urgent tasks.
+
+        :param model: Required. Gemini model name to process requests.
+        :param source: Required. Requests that will be sent to chosen model.
+            Can be in format of Inline requests or file name.
+        :param create_batch_job_config: Optional. Configuration parameters for batch job.
+        """
+        client = self.get_genai_client()
+        resp = client.batches.create(
+            model=model,
+            src=source,
+            config=create_batch_job_config,
+        )
+        return resp
+
+    def delete_batch_job(
+        self,
+        job_name: str,
+    ):
+        """
+        Delete batch job using Gemini Batch API.
+
+        :param job_name: Required. Batch job name.
+        """
+        client = self.get_genai_client()
+        client.batches.delete(name=job_name)
+
+    def cancel_batch_job(
+        self,
+        job_name: str,
+    ):
+        """
+        Cancel batch job using Gemini Batch API.
+
+        :param job_name: Required. Batch job name.
+        """
+        client = self.get_genai_client()
+        client.batches.cancel(
+            name=job_name,
+        )
+
+    def create_embeddings(
+        self,
+        model: str,
+        source: dict | str,
+        create_embeddings_config: CreateBatchJobConfig | dict | None = None,
+    ) -> BatchJob:
+        """
+        Create batch job for embeddings using Gemini Batch API to process large-scale, non-urgent tasks.
+
+        :param model: Required. Gemini model name to process requests.
+        :param source: Required. Requests that will be sent to chosen model.
+            Can be in format of Inline requests or file name.
+        :param create_embeddings_config: Optional. Configuration parameters for embeddings batch job.
+        """
+        client = self.get_genai_client()
+        input_type = "inlined_requests"
+
+        if isinstance(source, str):
+            input_type = "file_name"
+
+        self.log.info("Using %s to create embeddings", input_type)
+
+        resp = client.batches.create_embeddings(
+            model=model,
+            src={input_type: source},
+            config=create_embeddings_config,
+        )
+        return resp
+
+    def upload_file(self, path_to_file: str, upload_file_config: dict | None = None):
```

Comment:
Could we add return value type as we have in Client https://github.com/googleapis/python-genai/blob/main/google/genai/files.py#L410 ?

Reply:
done